woocommerce/woocommerce-gateway-stripe
# Skip Apple Pay registration for accounts from India

Fixes #2566

## Changes proposed in this Pull Request:

- Bail out before triggering Apple Pay registration if the account country is India.

## Testing instructions

**Confirming the registration is triggered with a supported account**
1. Connect the Stripe plugin to a test account, not from India. The one you probably have already would do
2. Add some fake live keys (the domain registration isn't triggered if they're empty). Stripe plugin Settings page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings` -> General -> Edit account keys -> Live -> Enter `pk_live_1234` for the live key and `sk_live_1234` for the secret key
3. Ensure test mode is enabled. Stripe plugin Settings page -> General -> Enable test mode
4. Ensure logging is enabled. Stripe plugin Settings page -> Advanced settings -> Check "Log error messages"
5. Disable Apple Pay / Google Pay Express checkouts, save, enable them again, and save. Stripe plugin Payment Methods page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods` -> Express checkouts -> Apple Pay / Google Pay
6. Go to Stripe's log file. `/wp-content/uploads/wc-logs/woocommerce-gateway-stripe-{date}.log`
7. Confirm there's an entry with the following
```
====Start Log====
Error: Unable to verify domain - Invalid API Key provided: sk_live_1234
====End Log====
```

**Confirming the registration isn't triggered with an account from India**
1. Create a test Stripe account with India as its country (or ping me for some creds)
2. Connect the Stripe plugin to this account. Stripe plugin Settings page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings` -> General -> Edit account keys -> Test -> Enter the keys
3. Disable Apple Pay / Google Pay Express checkouts, save, enable them again, and save. Stripe plugin Payment Methods page `{site}/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=methods` -> Express checkouts -> Apple Pay / Google Pay
4. Go to Stripe's log file. `/wp-content/uploads/wc-logs/woocommerce-gateway-stripe-{date}.log`
5. Confirm no entry for domain verification was added

---


**Post merge**
